internal/provider/service_account.go

```diff
@@ -0,0 +1,175 @@
+package provider
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/tabular-io/tabular-sdk-go/tabular"
+	"github.com/tabular-io/terraform-provider-tabular/internal/provider/util"
+)
+
+var (
+	_ resource.Resource                = &serviceAccountResource{}
+	_ resource.ResourceWithConfigure   = &serviceAccountResource{}
+	_ resource.ResourceWithImportState = &serviceAccountResource{}
+)
+
+type serviceAccountResource struct {
+	client *util.Client
+}
+
+func NewServiceAccountResource() resource.Resource {
+	return &serviceAccountResource{}
+}
+
+type serviceAccountResourceModel struct {
+	Id               types.String `tfsdk:"id"`
+	Name             types.String `tfsdk:"name"`
+	RoleId           types.String `tfsdk:"role_id"`
+	CredentialKey    types.String `tfsdk:"credential_key"`
+	CredentialSecret types.String `tfsdk:"credential_secret"`
+}
+
+func (r *serviceAccountResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	r.client = req.ProviderData.(*util.Client)
+}
+
+func (r *serviceAccountResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_account"
+}
+
+func (r *serviceAccountResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Tabular Service Account",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Credential ID",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "Service account name",
+				Required:    true,
+			},
+			"role_id": schema.StringAttribute{
+				Description: "Role ID",
+				Required:    true,
+			},
+			"credential_key": schema.StringAttribute{
+				Description: "Credential ID",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"credential_secret": schema.StringAttribute{
```

Comment:
Use case is to create service account and be able reference to use credentials in other resources?

Reply:
Yes, that's correct. The secret is only available at create time, so imported service accounts will not have a value for this attribute.